orthopairs/src/main/java/org/reactome/release/orthopairs/OrthopairFileGenerator.java

```diff
@@ -0,0 +1,93 @@
+    private static final Logger logger = LogManager.getLogger();
+    // Create source-target protein mapping file
+    public static void createProteinHomologyFile(String speciesKey, String sourceTargetProteinMappingFilename, JSONObject speciesJSON, Map<String,Set<String>> speciesProteinHomologs ) throws IOException {
+
+        logger.info("\tGenerating " + sourceTargetProteinMappingFilename);
+        File sourceTargetProteinMappingFile = new File(sourceTargetProteinMappingFilename);
+        if (sourceTargetProteinMappingFile.exists()) {
+            sourceTargetProteinMappingFile.delete();
+        }
+        sourceTargetProteinMappingFile.createNewFile();
+
+        List<String> sourceProteinIds = new ArrayList<>();
+        sourceProteinIds.addAll(speciesProteinHomologs.keySet());
+        Collections.sort(sourceProteinIds);
+        for (String sourceProteinId : sourceProteinIds) {
+            Set<String> targetProteinIds = speciesProteinHomologs.get(sourceProteinId);
+            targetProteinIds.remove("LDO");
+            String proteinOrthologLine = sourceProteinId.split("=")[1] + "\t";
+            List<String> cleanTargetProteinIds = new ArrayList<>();
+            for (String targetProteinId : targetProteinIds) {
+                cleanTargetProteinIds.add(targetProteinId.split("=")[1]);
+            }
+            Collections.sort(cleanTargetProteinIds);
+            proteinOrthologLine += String.join(" ", cleanTargetProteinIds) + "\n";
+            Files.write(Paths.get(sourceTargetProteinMappingFile.getPath()), proteinOrthologLine.getBytes(), StandardOpenOption.APPEND);
+        }
+    }
+
+    // Create target species gene-protein mapping file
+    public static void createSpeciesGeneProteinFile(String speciesKey, String targetGeneProteinMappingFilename, JSONObject speciesJSON, Map<String,Set<String>> speciesGeneProteinMap) throws IOException {
+
+        logger.info("\tGenerating " + targetGeneProteinMappingFilename);
+        File targetGeneProteinMappingFile = new File (targetGeneProteinMappingFilename);
+        if (targetGeneProteinMappingFile.exists()) {
+            targetGeneProteinMappingFile.delete();
+        }
+
+        boolean altIdMappingExists = false;
+        Map<String, Set<String>> altIdToEnsemblMap = new HashMap<>();
+        if (speciesJSON.get("alt_id_file") != null) {
+            logger.info("\tAlternate ID-Ensembl ID mapping required");
+            altIdToEnsemblMap = AlternateIdMapper.getAltIdMappingFile(speciesKey, speciesJSON.get("alt_id_file").toString());
+            altIdMappingExists = true;
+        }
+        targetGeneProteinMappingFile.createNewFile();
+
+        List<String> targetGeneProteinLines = new ArrayList<>();
+        for (String targetGeneId : speciesGeneProteinMap.keySet()) {
+            String[] geneSplit = targetGeneId.split("=");
+            String geneSource = geneSplit[0];
+            String geneId = geneSplit[geneSplit.length - 1];
```

Comment:
From this spot, the body of the for loop could be done like this to make the code more declarative:

```
String targetProteinIds = getTargetProteinsAsString(speciesGeneProteinMap.get(targetGeneId));
if (!targetProteinIds.isEmpty()) {
	if (!geneSource.startsWith("Ensembl") && altIdMappingExists(speciesJSON)) {
		targetGeneProteinLines.addAll(
			createGeneProteinLines(altIdToEnsemblMap.get(geneId), targetProteinIds)
		);
	} else {
		targetGeneProteinLines.add(createGeneProteinLine(geneId, targetProteinIds));
	}
}
```
With the private methods:
```
private static Set<String> createGeneProteinLines(Set<String> geneIds, String targetProteinIds) {
	return geneIds
		.stream()
		.map(geneId -> createGeneProteinLine(geneId, targetProteinIds))
		.collect(Collectors.toSet());
}

private static String createGeneProteinLine(String geneId, String targetProteinIds) {
	return geneId + "\t" + targetProteinIds + "\n";
}
```